docs/manual/src/interface/objects.md

```diff
@@ -0,0 +1,281 @@
+# Objects
+
+Objects are represented in the Rust world as a struct with an `impl` block containing methods.
+In the foreign-language bindings they would typically correspond to a class.
+
+In the interface definition, objects may have public methods but *must not have public fields*,
+because they are intended to be opaque to the foreign-language code. Like this:
```

Comment:
I initially wrote "I don't understand the justification for this limitation", but down in record.md I see that the public fields would make uniffi treat this as a record - which makes far more sense to me. Maybe update this para to reflect that?